app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -0,0 +1,25 @@
+package ch.epfl.favo.common;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.Query;
+
+public final class DatabaseWrapper {
+
+    private static FirebaseFirestore mFirestore;
+
+    public DatabaseWrapper() {
+        FirebaseFirestore.setLoggingEnabled(true);
+        this.mFirestore = FirebaseFirestore.getInstance();
```

Comment:
You must not instantiate a static field in the constructor. What would happen if *getDatabaseInstance* is called before the constructor? this has to be possible since it is a static method. 
That's a wrong use of static fields and methods.